rosbag2_cpp/test/rosbag2_cpp/test_multifile_reader.cpp

```diff
@@ -40,75 +42,165 @@ class MultifileReaderTest : public Test
     converter_factory_(std::make_shared<StrictMock<MockConverterFactory>>()),
     storage_serialization_format_("rmw1_format"),
     relative_path_1_("some_relative_path_1"),
-    relative_path_2_("some_relative_path_2")
+    relative_path_2_("some_relative_path_2"),
+    absolute_path_1_("/some/absolute/path1"),
+    storage_uri_(rcpputils::fs::temp_directory_path().string()),
+    default_storage_options_({storage_uri_, ""})
+  {}
+
+  virtual void init()
   {
+    auto metadata = get_metadata();
+
     auto topic_with_type = rosbag2_storage::TopicMetadata{
       "topic", "test_msgs/BasicTypes", storage_serialization_format_, ""};
     auto topics_and_types = std::vector<rosbag2_storage::TopicMetadata>{topic_with_type};
+    metadata.topics_with_message_count.push_back({topic_with_type, 10});
 
     auto message = std::make_shared<rosbag2_storage::SerializedBagMessage>();
     message->topic_name = topic_with_type.name;
 
+    auto storage_factory = std::make_unique<StrictMock<MockStorageFactory>>();
     auto metadata_io = std::make_unique<NiceMock<MockMetadataIo>>();
-    rosbag2_storage::BagMetadata metadata;
-    metadata.relative_file_paths.push_back(relative_path_1_);
-    metadata.relative_file_paths.push_back(relative_path_2_);
-    metadata.topics_with_message_count.push_back({topic_with_type, 10});
     ON_CALL(*metadata_io, read_metadata(_)).WillByDefault(Return(metadata));
     EXPECT_CALL(*metadata_io, metadata_file_exists(_)).WillRepeatedly(Return(true));
 
-    auto storage_factory = std::make_unique<StrictMock<MockStorageFactory>>();
     EXPECT_CALL(*storage_, get_all_topics_and_types())
     .Times(AtMost(1)).WillRepeatedly(Return(topics_and_types));
     ON_CALL(*storage_, read_next()).WillByDefault(Return(message));
     EXPECT_CALL(*storage_factory, open_read_only(_, _)).WillRepeatedly(Return(storage_));
 
     auto sequential_reader = std::make_unique<rosbag2_cpp::readers::SequentialReader>(
       std::move(storage_factory), converter_factory_, std::move(metadata_io));
-
     reader_ = std::make_unique<rosbag2_cpp::Reader>(std::move(sequential_reader));
   }
 
+  virtual ~MultifileReaderTest() = default;
+
+  virtual rosbag2_storage::BagMetadata get_metadata() const
+  {
+    rosbag2_storage::BagMetadata metadata;
+
+    metadata.relative_file_paths.push_back(relative_path_1_);
+    metadata.relative_file_paths.push_back(relative_path_2_);
+    metadata.relative_file_paths.push_back(absolute_path_1_);
```

Comment:
```suggestion
    metadata.relative_file_paths = {relative_path_1_, relative_path_2_, absolute_path_1_};
```
Unless you think the current way is cleaner.